### PR TITLE
JBEHAVE-1491 Add type conversion API into Parameters class

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: jbehave-core/src/main/java/org/jbehave/core/steps/Parameters.java [hunk 1]
```diff
@@ -27,4 +28,20 @@
...
+     *
+     * @param type The target type
+     * @return The object of type <T>
+     */
+    <T> T mapTo(Class<T> type);
+
+    /**
+     * Maps parameters to the specified type
+     *
+     * @param The target type
```
Comment (valfirst) [inline jbehave-core/src/main/java/org/jbehave/core/steps/Parameters.java]:
```suggestion
     * @param type The target type
```

File: jbehave-core/src/main/java/org/jbehave/core/steps/ConvertedParameters.java [hunk 2]
```diff
@@ -47,6 +60,81 @@ public ConvertedParameters(Map<String, String> values, ParameterConverters param
...
 
+    @Override
+    public <T> T mapTo(Class<T> type) {
+        return mapTo(type, Collections.emptyMap());
+    }
+
+    @Override
+    public <T> T mapTo(Class<T> type, Map<String, String> fieldNameMapping) {
+        try {
+            T instance = type.newInstance();
```
Comment (valfirst) [inline jbehave-core/src/main/java/org/jbehave/core/steps/ConvertedParameters.java]:
```suggestion
            T instance = type.getDeclaredConstructor().newInstance();
```

File: jbehave-core/src/main/java/org/jbehave/core/steps/ConvertedParameters.java [hunk 3]
```diff
@@ -47,6 +60,81 @@ public ConvertedParameters(Map<String, String> values, ParameterConverters param
...
+        Map<String, Field> mappedFields = new HashMap<>();
+        List<String> unmappableFields = new ArrayList<>();
+
+        fields.forEach((fieldName, fieldWrapper) -> {
+            if (fieldWrapper.isPresent()) {
+                mappedFields.put(fieldName, fieldWrapper.get());
+                return;
+            }
+            unmappableFields.add(fieldName);
+        });
```
Comment (valfirst) [inline jbehave-core/src/main/java/org/jbehave/core/steps/ConvertedParameters.java]:
```suggestion
        Map<String, Field> mappedFields = new HashMap<>();
        List<String> unmappableFields = new ArrayList<>();
        for (String fieldName : fieldNames)
        {
            Optional<Field> fieldWrapper = findField(type, fieldName, fieldNameMapping);
            if (fieldWrapper.isPresent()) {
                mappedFields.put(fieldName, fieldWrapper.get());
            }
            else {
                unmappableFields.add(fieldName);
            }
        }

        if (unmappableFields.isEmpty()) {
            return mappedFields;
        }
```

**Review by valfirst — CHANGES_REQUESTED** [review]

File: jbehave-core/src/main/java/org/jbehave/core/steps/ConvertedParameters.java (hunk 2 above)
Comment (uarlouski) [reply jbehave-core/src/main/java/org/jbehave/core/steps/ConvertedParameters.java]: just for reference https://docs.oracle.com/javase/9/docs/api/java/lang/Class.html#newInstance--

**Review by valfirst — APPROVED** [review]

**Comment by uarlouski** [comment]
In master https://github.com/jbehave/jbehave-core/commit/a295e2121ec874c51c5e06c94bdba682c4ba2937

**Pull request closed without merging**